File: backend/app/schemas/proposal.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from .family_member import FamilyMemberResponse


class ProposalCreate(BaseModel):
    proposed_date: datetime
    message: str | None = None


class ProposalRespondRequest(BaseModel):
    response: Literal["accepted", "rejected"]
    message: str | None = None
    counter_date: datetime | None = None


class ProposalResponseDetail(BaseModel):
    id: int
    member: FamilyMemberResponse
    response: str
    counter_proposal_id: int | None
    message: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class ProposalDetail(BaseModel):
    id: int
    todo_id: int
    proposer: FamilyMemberResponse
    proposed_date: datetime
    message: str | None
    status: str
    responses: list[ProposalResponseDetail]
    created_at: datetime

    model_config = {"from_attributes": True}


class PendingProposalDetail(BaseModel):
    id: int
    todo_id: int
    todo_title: str
    proposer: FamilyMemberResponse
    proposed_date: datetime
    message: str | None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: obsolete/ios/Familienkalender/Models/Proposal.swift
import Foundation

struct ProposalCreate: Codable {
    let proposedDate: String
    var message: String?
}

struct ProposalRespondRequest: Codable {
    let response: String
    var message: String?
    var counterDate: String?
}

struct ProposalDetail: Codable, Identifiable {
    let id: Int
    let todoId: Int
    let proposer: FamilyMemberResponse
    let proposedDate: String
    let message: String?
    let status: String
    let responses: [ProposalResponseDetail]
    let createdAt: String
}

struct ProposalResponseDetail: Codable, Identifiable {
    let id: Int
    let member: FamilyMemberResponse
    let response: String
    let counterProposalId: Int?
    let message: String?
    let createdAt: String
}

struct PendingProposalDetail: Codable, Identifiable {
    let id: Int
    let todoId: Int
    let todoTitle: String
    let proposer: FamilyMemberResponse
    let proposedDate: String
    let message: String?
    let status: String
    let createdAt: String
}
